std/stdio.d

```diff
@@ -1769,6 +1769,7 @@ is recommended if you want to process a complete file.
      * $(REF formattedRead, std,_format).
      */
     uint readf(Data...)(in char[] format, Data data)
+        if (allSatisfy!(isPointer, Data))
```

Comment:
This style is all over Phobos.

Could you please write a few simple rules on constraints for the D style guide? Because right now I'm hesitant to hold up any PR over this if we don't have clear guidelines.